apps/avatax/src/modules/calculate-taxes/use-case/calculate-taxes.use-case.test.ts

```diff
@@ -175,4 +208,57 @@ describe("CalculateTaxesUseCase", () => {
     // Expect any error to be attached. We dont yet specify errors so these tests will be added later
     expect(error.errors![0]).toBeInstanceOf(Error);
   });
+
+  it("Uses AutomaticallyDistributedDiscountsStrategy to calculate discounts if present", async () => {
+    mockGetAppConfig.mockImplementationOnce(() => ok(getMockedAppConfig()));
+
+    mockedAvataxClient.createTransaction.mockResolvedValueOnce({ lines: [] });
+
+    const payload = getPayloadWithDiscounts();
+
+    await instance.calculateTaxes(payload, getMockAuthData());
+
+    expect(mockedAvataxClient.createTransaction).toHaveBeenCalledOnce();
+
+    expect(mockedAvataxClient.createTransaction).toHaveBeenCalledWith({
+      model: {
+        addresses: {
+          shipFrom: {
+            city: "test",
+            country: "test",
+            line1: "test",
+            postalCode: "10111",
+            region: "NY",
+          },
+          shipTo: {
+            city: "",
+            country: "PL",
+            line1: "",
+            line2: "",
+            postalCode: "",
+            region: "",
+          },
+        },
+        commit: false,
+        companyCode: "test",
+        currencyCode: "PLN",
+        customerCode: "0",
+        date: expect.any(Date),
+        discount: 10.1,
+        entityUseCode: "",
+        lines: [
+          {
+            amount: 100,
+            description: undefined,
+            discounted: true,
```

Comment:
for previty and to expose what is ACTUAL assertion you can use expect.objectContaining to only check fields that are important in this case.

non blocking